### PR TITLE
Added the Set GP Layer Attributes and Set GP Stroke Attributes nodes.

### PR DESCRIPTION
**Pull request by 3DSinghVFX**
I have added the _Set GP Layer Attributes_ and _Set GP Stroke Attributes_ nodes which allow to set all the attributes together of a Layer and Stroke,
Screenshot from 2020-02-11 01-21-42

_More discussion for these nodes is here: https://github.com/JacquesLucke/animation_nodes/pull/1307#issuecomment-584209257_

### CODE REVIEW

**Comment by OmarEmaraDev** [comment]
The input attributes should be independent of the `useLayerList`. In other words, we should allow a mix between single and list attributes.

You may also use a virtual list for layers and strokes. And you shouldn't create virtual lists if they wre not going to be used.

Also, remove the set nodes we added in #1307.

**Comment by 3DSinghVFX** [comment]
> The input attributes should be independent of the `useLayerList`. In other words, we should allow a mix between single and list attributes.

It is added.

> You may also use a virtual list for layers and strokes. And you shouldn't create virtual lists if they wre not going to be used.

It is added.
 
> Also, remove the set nodes we added in #1307.

Removed.

File: animation_nodes/nodes/gpencil/set_gp_layer_attributes.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... data_structures import GPLayer, VirtualPyList, VirtualDoubleList, VirtualLongList
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/gpencil/set_gp_layer_attributes.py]: This import is not needed.

File: animation_nodes/nodes/gpencil/set_gp_layer_attributes.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            socket.useIsUsedProperty = True
+            socket.isUsed = False
+
+    def getExecutionCode(self, required):
+        s = self.inputs
+        isName = s[1].isUsed
+        isBlendMode = s[2].isUsed
+        isOpacity = s[3].isUsed
+        isPassIndex = s[4].isUsed
+        if self.useLayerList or self.useNameList or self.useBlendModeList or self.useOpacityList or self.usePassIndexList:
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/gpencil/set_gp_layer_attributes.py]: Use the built-in `any()` function.

File: animation_nodes/nodes/gpencil/set_gp_layer_attributes.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            ("Layer", "layers"), ("Layers", "layers")), dataIsModified = True)
+        self.newInput(VectorizedSocket("Text", "useNameList",
+            ("Name", "layerNames"), ("Names", "layerNames")), value = 'AN-Layer')
+        self.newInput(VectorizedSocket("Text", "useBlendModeList",
+            ("Blend Mode", "blendModes"), ("Blend Modes", "blendModes")), value = 'REGULAR', hide = True)
+        self.newInput(VectorizedSocket("Float", "useOpacityList",
+            ("Opacity", "opacities"), ("Opacities", "opacities")), value = 1, hide = True)
+        self.newInput(VectorizedSocket("Integer", "usePassIndexList",
+            ("Pass Index", "passIndices"), ("Pass Indices", "passIndices")), value = 0, hide = True)
+        self.newOutput(VectorizedSocket("GPLayer", ["useLayerList", "useNameList", "useBlendModeList",
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/gpencil/set_gp_layer_attributes.py]: Put a new line before `self.newOutput`. Try to organize this a bit more, maybe:
```python
self.newOutput(VectorizedSocket("GPLayer",
    ["useLayerList", "useNameList", "useBlendModeList", "useOpacityList", "usePassIndexList"],
    ("Layer", "outLayer"), ("Layers", "outLayers")))
```

File: animation_nodes/nodes/gpencil/set_gp_layer_attributes.py (hunk 2 above)
Comment (3DSinghVFX) [reply animation_nodes/nodes/gpencil/set_gp_layer_attributes.py]: Done.

File: animation_nodes/nodes/gpencil/set_gp_layer_attributes.py (hunk 1 above)
Comment (3DSinghVFX) [reply animation_nodes/nodes/gpencil/set_gp_layer_attributes.py]: Done.

File: animation_nodes/nodes/gpencil/set_gp_layer_attributes.py (hunk 3 above)
Comment (3DSinghVFX) [reply animation_nodes/nodes/gpencil/set_gp_layer_attributes.py]: Done.

**Comment by 3DSinghVFX** [comment]
Is it possible to set that in the _newInput_?

**Comment by OmarEmaraDev** [comment]
Yes. You can pass `minValue` if I remember correctly.

**Comment by 3DSinghVFX** [comment]
Okay.

**Comment by OmarEmaraDev** [comment]
Opacity needs a maximum of 1 as well.

**Comment by 3DSinghVFX** [comment]
It is done.

**Comment by 3DSinghVFX** [comment]
Should I update the CHANGELOG.md?

**Comment by OmarEmaraDev** [comment]
Yes. Please do that before submitting the pull request in the future.

**Comment by 3DSinghVFX** [comment]
> Yes. Please do that before submitting the pull request in the future.

Okay.

**Pull request merged**